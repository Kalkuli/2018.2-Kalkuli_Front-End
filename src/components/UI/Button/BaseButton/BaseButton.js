import React from 'react'
import './BaseButton.scss'

const BaseButton = (props) => {
  
  let styles = ["base-button"]
  if(props.type === 'confirm')
    styles.push('confirm')
  else if(props.type === 'cancel')
    styles.push('cancel')

  return (
    <a className={styles.join(' ')}>
      {props.children}
    </a>
  )
}

export default BaseButton