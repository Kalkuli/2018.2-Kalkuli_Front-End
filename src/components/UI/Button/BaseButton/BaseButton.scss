@import "../../../../assets/sass/main.scss";

.base-button {
  font-size: 2rem;
  border-radius: 7px;
  user-select: none;
  cursor: pointer;
  transition: all .3s;
  width: 300px;
  padding: 1rem;

  @media screen and (max-width: 980px) {
    font-size: 1.3rem;
  }

  &--small {
    width: 10em;
    font-size: 1.5rem;
  }
}

.confirm {
  text-align: center;
  color: $color-white;
  background-color: $color-dark-blue;
}

.cancel {
  text-align: center;
  color: $color-dark-blue;
  border: 1px solid $color-dark-blue;
}

.choose-file {
  font-family: 'Alfa Slab One', cursive;
  border: 1px solid $color-light-blue;
  font-weight: 100;
  &:hover {
    box-shadow: 0 1px 1px $color-black;
    transform: translateY(-0.3rem);
  }
  
  &:active {
    box-shadow: 0 0 0 $color-black;
    transform: translateY(0);
  }
}