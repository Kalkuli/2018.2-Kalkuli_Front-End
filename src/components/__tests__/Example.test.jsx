import React from 'react';
import { shallow } from 'enzyme';
import BaseButton from '../UI/Button/BaseButton/BaseButton';

// If snapshot is needed
import renderer from 'react-test-renderer';

test('Text example', () => {

    // Shallow is needed to enssure that the component is not modified by any modification
    // made at its' child components 
    const wrapper = shallow(<BaseButton/>);
    const element = wrapper.getElement();
    
    // To print elements, use:
    // console.log(element)
    
    // Test's Assetion
    expect(element.props.className).toBe('base-button');
    
    // You can also spashot elements, snapshotting is used for ensure that the component's  
    // "apearence" does not change. Read more at: https://jestjs.io/docs/en/snapshot-testing 
    const tree = renderer.create(<BaseButton/>).toJSON();
    expect(tree).toMatchSnapshot();

});