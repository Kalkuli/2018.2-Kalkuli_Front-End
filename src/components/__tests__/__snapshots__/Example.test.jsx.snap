// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test example 1`] = `
<a
  className="base-button"
  onClick={undefined}
/>
`;
